src/Http/Routing/src/Matching/AcceptsMatcherPolicy.cs

```diff
@@ -236,9 +236,13 @@ public IReadOnlyList<PolicyNodeEdge> GetEdges(IReadOnlyList<Endpoint> endpoints)
             edges.Add(string.Empty, anyEndpoints.ToList());
         }
 
-        return edges
-            .Select(kvp => new PolicyNodeEdge(kvp.Key, kvp.Value))
-            .ToArray();
+        var result = new PolicyNodeEdge[edges.Count];
+        var index = 0;
+        foreach (var kvp in edges)
+        {
+            result[index++] = new PolicyNodeEdge(kvp.Key, kvp.Value);
```

Comment:
nit: "one" operation per line. Move the increment to a separate line, makes it more obvious what is happening and less likely to introduce bugs.